src/cache/event.rs

```diff
@@ -97,9 +97,12 @@ impl CacheUpdate for ChannelDeleteEvent {
         };
 
         // Remove the cached messages for the channel.
-        cache.messages.remove(&self.channel.id());
-
-        None
+        match cache.messages.remove(&self.channel.id()) {
+            Some((_, messages)) => {
+                Some(messages.values().map(|msg| msg.to_owned()).collect::<Vec<_>>())
+            },
+            None => None,
+        }
```

Comment:
```suggestion
        cache
            .messages
            .remove(&self.channel.id())
            .map(|(_, messages)| messages.values().cloned().collect())
```
You can use `Option::map` and `Iterator::cloned` here, and `collect` can actually infer its return type in this case.